### PR TITLE
Remove claim module

### PR DESCRIPTION
**Pull request by faddat**
Closes: #903 

This PR:

* Removes the claim module
* changes the module path of osmosis to github.com/osmosis-labs/osmosis/v8

### CODE REVIEW

**Review by alexanderbez — APPROVED** [review]

**Review by xBalbinus — APPROVED** [review]

**Comment by sunnya97** [comment]
Should we prune all the claim module's state?

**Comment by ValarDragon** [comment]
Awesome! In the upgrade handler logic, https://github.com/osmosis-labs/osmosis/blob/main/app/app.go#L424-L432
we need to add a store upgrade for v8, that makes the claim module deleted. That will remove this from state.

We should check that IBC channels still work gracefully across the upgrade, but I suspect it should be fine, since module additions work just fine.

**Comment by alexanderbez** [comment]
@faddat mind resolving the conflicts and we can get this merged pls?

**Comment by ValarDragon** [comment]
Lets get this merged after #1304 , and do the deletion from the store list in this PR as well

**Comment by faddat** [comment]
Happy to

**Review by ValarDragon — APPROVED** [review]
Added the claims deletion logic!

Now LGTM, thanks for making this PR!

File: cmd/osmosisd/cmd/balances_from_state_export.go.history [hunk 1]
```diff
@@ -257,34 +255,6 @@ Example:
 				snapshotAccs[address] = acc
 			}
 
-			claimGenesis := claimtypes.GenesisState{}
```
Comment (ValarDragon) [inline cmd/osmosisd/cmd/balances_from_state_export.go.history:260]: Oh this should not have been moved to a history file. We need to undo this

File: cmd/osmosisd/cmd/balances_from_state_export.go.history [hunk 2]
```diff
@@ -257,34 +255,6 @@ Example:
...
-					)
-				}
-
-				for action := range claimtypes.Action_name {
-					if record.ActionCompleted[action] == false {
-						acc.UnclaimedAirdrop = acc.UnclaimedAirdrop.Add(claimablePerAction...)
-					}
-				}
-
-				snapshotAccs[address] = acc
```
Comment (ValarDragon) [inline cmd/osmosisd/cmd/balances_from_state_export.go.history:285]: Oh no, this was an error as well :/

File: cmd/osmosisd/cmd/balances_from_state_export.go.history [hunk 3]
```diff
@@ -313,8 +283,6 @@ Example:
 					Add(sdk.NewCoin(appparams.BaseCoinUnit, account.Staked)).
 					Add(sdk.NewCoin(appparams.BaseCoinUnit, account.UnbondingStake)).
 					Add(account.Bonded...).
-					Add(account.UnclaimedAirdrop...)
-				snapshotAccs[addr] = account
```
[the remark this answers is not in the document]
Comment (alexanderbez) [reply cmd/osmosisd/cmd/balances_from_state_export.go.history:317]: What do you mean error? I removed it in the backport branch. Do we need to add it back

**Pull request merged**